src/ck-core/cklocation.h

```diff
@@ -651,10 +650,13 @@ class CkLocMgr : public IrrGroup
   // int deliverMsg(CkMessage *m, CkArrayID mgr, const CkArrayIndex &idx, CkDeliver_t
   // type, int opts = 0);
   int deliverMsg(CkArrayMessage* m, CkArrayID mgr, CmiUInt8 id, const CkArrayIndex* idx,
-                 CkDeliver_t type, int opts = 0);
+                 CkDeliver_t type, bool firstAttempt = true, int opts = 0);
```

Comment:
Does true make sense as the default (for all of these defaults)? I would think that the first send is more of the exception? Just a minor thought though.

Reply:
My tracing through the function call paths says yes, but I agree that it's a brittle solution that we should ideally solve in another way.